### PR TITLE
feat(sdk): update dependency flutter to v3.27.0 and dart to >=3.6.0 <4.0.0

### PR DESCRIPTION
**Pull request by jackson-chandler-basys**
- Convert `DialogTheme` to `DialogThemeData`
- Convert `TabBarTheme` to `TabBarThemeData`
- Convert `DialogTheme` to `DialogThemeData`
- Upgrade package version
- Upgrade flutter version in `.fvm` and `pubspec.yaml`
- Bump package version to `6.6.7`

Hey Zeshuaro, big fan of your work in this package. The recent upgrade to Flutter version `3.27.0` broke my project, so I went ahead and made the changes in the PRs from the renovate bot and fixed the TODOs and errors that came with that upgrade. I hope this satisfies your projects contribution requirements.

Thanks

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -1,11 +1,11 @@
 name: json_theme_plus
 description: A library to dynamically generate a ThemeData object from a JSON file or dynamic map object
 homepage: https://github.com/zeshuaro/json_theme_plus
-version: 6.6.6
+version: 6.6.7
```
Comment (zeshuaro) [inline pubspec.yaml:4]: There's no need to update this. It'll be automatically updated when the package gets released.

**Review by zeshuaro — CHANGES_REQUESTED** [review]
Thanks for your help to support the latest Flutter version. Just one minor feedback if you can address that then we can get your changes merged in 🙂

File: pubspec.yaml (hunk 1 above)
Comment (jackson-chandler-basys) [reply pubspec.yaml:4]: Great! I'll get that changed real quick

Comment (jackson-chandler-basys) [reply pubspec.yaml]: @zeshuaro changes are in, thanks!

**Comment by zeshuaro** [comment]
Can you also run `dart format .` to get all the files formatted for [linting](https://github.com/zeshuaro/json_theme_plus/actions/runs/12316632999/job/34384263931?pr=106) to pass?

**Comment by jackson-chandler-basys** [comment]
I have run the command and pushed the changes!

**Review by zeshuaro — DISMISSED** [review]

**Comment by zeshuaro** [comment]
There's some errors from `flutter analyze`, can you also fix those?

**Comment by jackson-chandler-basys** [comment]
Those deprecations were a bit out of the scope for the changes I made and I wasn't sure about the exact migration, but I can give it a shot!

**Comment by jackson-chandler-basys** [comment]
@zeshuaro looks like the `Color.value` deprecation still have a lot of moving parts and changing to something else will require additional code to fix. Do you still want to go down that route or do you want to hold off and just put an ignore in the files for this specific deprecation?
[Flutter issue in question](https://github.com/flutter/flutter/issues/160184)

**Comment by zeshuaro** [comment]
I see, can you just disable the linter inline with a comment for now to keep the linter happy?

**Comment by jackson-chandler-basys** [comment]
@zeshuaro done and done!

**Review by zeshuaro — APPROVED** [review]

**Comment by jackson-chandler-basys** [comment]
Thank you so much!

**Pull request merged**